platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java

```diff
@@ -105,9 +108,38 @@ public synchronized void setContextPolicy(String path, ContextPolicy contextPoli
         if (contextPolicy == null) {
             throw new IllegalArgumentException("Context policy cannot be null.");
         }
+        
         LOGGER.debug("setContextPolicy called with path = {}", path);
-        policyStore.put(path, contextPolicy);
 
+        //gather all context realms, authorization types, & required attributes
+        Map<String, String> contextsToRealms = new HashMap<String, String>();
+        Map<String, List<ContextAttributeMapping>> contextsToAttrs = new HashMap<>();
+        Map<String, List<String>> contextsToAuths = new HashMap<>();
+
+        for (String contextPath : policyStore.keySet()) {
```

Comment:
`entrySet` iteration would be a bit easier here as you're using the value multiple times.